src/MongoDB.Driver.Core/Core/Clusters/ClusterDescription.cs

```diff
@@ -240,29 +274,5 @@ public ClusterDescription WithType(ClusterType value)
         {
             return _type == value ? this : new ClusterDescription(_clusterId, _connectionMode, value, _servers);
         }
-
-        // private methods
-        private TimeSpan? CalculateLogicalSessionTimeout(IEnumerable<ServerDescription> servers)
-        {
-            TimeSpan? logicalSessionTimeout = null;
-
-            foreach (var server in servers)
-            {
-                if (server.IsDataBearing)
```

Comment:
Note that this was not taking direct ConnectionMode into account. See revised method (now static) above.